### ISSUE TITLE
Feature request - Deploy to NAS as app or docker

### ISSUE THREAD

**Issue by Darth-Coin** [getumbrel/umbrel#747]
There are many users and in special companies that are using NAS (network area storage) like QNAP and Synology.
Are great machines (linux based) and some versions come with Docker deploy and even VM.
It will be nice to have like an Umbrel app to be able to deploy it directly into a NAS machine and have own node ready.
Examples here https://www.qnap.com/en/app_center/ or using built in Docker.
I suggest that Umbrel devs to take a look and keep in consideration for future deploys.

**Comment by mariuspop86** [comment]
You can already do this. I'm using this, on a Synology NAS. Running umbrel on a Ubuntu virtual machine.

**Comment by algotechio** [comment]
The VM solution works on synology.  The only issue is that even with 1TB SSD, I am almost running out of space (at 81%).
I would like move the bitcoin blockchain data to other HDD drive on synology, but then the only way to do this is to mount the HDD via NFS (I don't like this solution).
In terms of performance, bitcoin blocks on HDD should work OK as the apps are communicating with indexers, such as electrs and dojo, so having bitcoin blocks sitting on HDD should be no problem.

In case of synology, the only OS dependency missing is fswatch (jq, rsync, curl are present) and unfortunately, there is no easy way to install it (I would also worry about the stability of the system and the persistence between synology OS updates.) 
I think the fswatch should work inside containers ([Inotify in Containers](https://william-yeh.net/post/2019/06/inotify-in-containers/)), so it would be really nice to containerize the backup functionality (it seems that it's the only place where it's needed).  It would be interesting to hear comments of @lukechilds / @mayankchhabra about the direction.

The other potential issue of running umbrel directly on synology, is [setup-iptables](https://github.com/getumbrel/umbrel/blob/master/scripts/umbrel-os/status-server/setup-iptables), as this would create a conflict with the admin interface of the synology server.  I am not sure the reason for modifying the IP tables.  Other than that, the service can run already run on separate port using `NGINX_PORT`

I think it would be really nice to be able to run umbrel NASes without using VMs.  In order to do that, it seems to me that
1. It would be great to dockerize every single script.  This would avoid installing dependencies.
2. Do minimum direct OS scripting.  It would be great if one day we could just run `docker-compose up -d` to startup umbrel

I'd be happy to help maybe with the dockerization of the backups, if there is a chance of having this merged back.

**Comment by lukechilds** [comment]
Hey @algotechio thanks for taking the time to write this up an for offering to help!

>In terms of performance, bitcoin blocks on HDD should work OK as the apps are communicating with indexers, such as electrs and dojo,

It's definitely usable but it will affect performance quite a lot. Both electrs and Dojo forward requests on to Bitcoin Core. Most users repot a few days to do IBD on an SSD and a few months on an HDD.

>it would be really nice to containerize the backup functionality

Yeah that's planned, we haven't quite figured out the final architecture for apps yet so that's needs solidifying before we start refactoring other areas.

>I am not sure the reason for modifying the IP tables.

We recently shipped an alternative server that runs before Umbrel and Docker containers have started to update the user on the boot process. The iptables rule transparently switches port 80 between the boot server and the Umbrel server once it's online. This is only used by the systemd services in Umbrel OS, you don't need to worry about it on a custom install.

>I think it would be really nice to be able to run umbrel NASes without using VMs.

>It would be great if one day we could just run `docker-compose up -d` to startup umbrel

Yeah this was exactly our thinking originally but the more we've gone forward the more we've realised it's probably not going to work well. Umbrel already has a few responsibilities that require access to the OS and can't be done reliably from inside a contianer. e.g power cycling the host, installing packages on the host, running services on the host, managing containers, updating iptables rules etc. We've done our best to keep separation of concerns, but the more we build the more we realise Umbrel really does need complete control of the host to function properly. This is likely to increase as we implement stricter app sandboxing behaviour too.

Already the difference between users running Umbrel OS and running custom Umbrel installs is causing quite a bit of development burden. Every piece of logic that does something on the host needs to be conditionally checked to see if it's safe to do on a custom install. Consider that 99% of users are running Umbrel OS on a Raspberry Pi, it means a large amount of development time is spent on supporting a very uncommon use case.

The best way forward would probably be for us to drop the idea of installing Umbrel as a standalone service and instead release some sort of platform agnostic Umbrel OS image that can be run on a range of different hardware and VMs. That way, we keep development efficient and can focus on a single deployment target, but users still have the freedom to deploy Umbrel on whatever hardware they wish, including existing servers via a VM.

I can't make any commitments as to exactly when this would happen, it's not our immediate priority right now, but I think this architecture makes most sense for us to move towards going forwards.
